src/Internal/Payment/State/InitialState.php

```diff
@@ -187,12 +194,34 @@ protected function populate_context_from_order() {
 			)
 		);
 		$context->set_level3_data( $this->level3_service->get_data_from_order( $order_id ) );
+	}
 
-		// Customer management involves a remote call.
-		$customer_id = $this->customer_service->get_or_create_customer_id_from_order(
-			$context->get_user_id(),
-			$this->order_service->_deprecated_get_order( $order_id )
-		);
+	/**
+	 * Parse and manage customer details.
+	 * This includes the update/creation of a customer and link it to the order.
+	 *
+	 * @throws Order_Not_Found_Exception In case the order could not be found.
+	 * @throws API_Exception
+	 * @throws Exception
+	 */
+	protected function manage_customer_details() {
+		$context       = $this->get_context();
+		$user_id       = $context->get_user_id();
+		$user          = get_user_by( 'id', $user_id );
+		$customer_data = WC_Payments_Customer_Service::map_customer_data( $this->order_service->_deprecated_get_order( $context->get_order_id() ), new WC_Customer( $user_id ) );
```

Comment:
Can we move most of this logic out of the state, and into the service? As far as the payment process is concerned, there is `( user_id, order|wc_customer )  => customer_id`, and that can apply to other situations. IMO this could live in the `get_or_create_customer_id_from_order` I created for the new payment process (and is otherwise unused right now).

Reply:
This is refactored and simplified (look for previous comments). Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7640/commits/e86e5ef2e21cbe1588a18de580827515557ac946